### PR TITLE
Fix ambiguous encoding #351

### PR DESCRIPTION
**Pull request by ideasman42**

### CODE REVIEW

File: src/gpodder/extensions.py [hunk 1]
```diff
@@ -289,10 +289,10 @@ def load_extension(self):
             return
 
         basename, extension = os.path.splitext(os.path.basename(self.filename))
-        fp = open(self.filename, 'r')
+        fp = open(self.filename, 'rb')
```
Comment (elelay) [inline src/gpodder/extensions.py]: Why open `.py` as rb?

File: src/gpodder/extensions.py [hunk 2]
```diff
@@ -233,7 +233,7 @@ def _load_metadata(self, filename):
         if not filename or not os.path.exists(filename):
             return {}
 
-        extension_py = open(filename).read()
+        extension_py = open(filename, encoding='utf-8').read()
```
Comment (elelay) [inline src/gpodder/extensions.py]: Looks good to me.

**Comment by ideasman42** [comment]
Right, and if people want to share extensions it's best to stick to one encoding.
(maybe it could be detected, although I'm not sure it's worth it).

Note that another way to handle this is using Pythons AST to parse the information instead of reading as text. But think this isn't necessary.

**Comment by elelay** [comment]
@ideasman42 can you explain why you change open 'r' to open 'rb'?

**Comment by ideasman42** [comment]
@elelay because for loading as data for a module, there is no need to use string/encoding.
It can load as data instead (It's a little faster and less complicated)

**Comment by thp** [comment]
This is not the way to go. The right filesystem encoding needs to be used. I'll comment further in #351.

**Pull request closed without merging**